LempelZivWelch/DataHandler/DecodeDataHandler.cs

```diff
@@ -0,0 +1,102 @@
+﻿// <copyright file="DecodeDataHandler.cs" company="Roman Levashev">
+// Copyright (c) Roman Levashev. All rights reserved.
+// Licensed under the MIT License.
+// </copyright>
+
+namespace DataHandler;
+
+/// <summary>
+/// Handles decoding of compressed data by reading bits from an input stream and writing decoded sequences to an output stream.
+/// </summary>
+/// <param name="inputFileStream">The input stream containing compressed data.</param>
+/// <param name="outputFileStream">The output stream to write decoded data to.</param>
+/// <param name="bufferSize">The size of the buffer used for reading and writing operations.</param>
+/// <param name="chunkSize">The number of bits to read for each code during decoding.</param>
+public class DecodeDataHandler(FileStream inputFileStream, FileStream outputFileStream, int bufferSize, int chunkSize) : InputDataHandler(bufferSize, inputFileStream)
+{
+    private readonly FileStream outputFileStream = outputFileStream;
+
+    private int bufferSize = bufferSize;
+
+    private List<byte> outputBuffer = [];
+
+    private Queue<bool> bitBuffer = [];
+
+    /// <summary>
+    /// Gets a value indicating whether both the file read operation is complete and the bit buffer contains less data than the chunk size.
+    /// </summary>
+    public override bool IsFileReadComplete
+    {
+        get => base.IsFileReadComplete && this.bitBuffer.Count < this.ChunkSize;
+    }
```

Comment:
```suggestion
    public override bool IsFileReadComplete => base.IsFileReadComplete && this.bitBuffer.Count < this.ChunkSize;
```
То же самое, но короче